### PR TITLE
feat: implement API for Compute/Verify MAC operations

### LINKED ISSUE #39: Implement API for computing/verifying MAC

### PR DESCRIPTION
**Pull request by aholovko**
Closes #39

Signed-off-by: Andriy Holovko <andriy.holovko@gmail.com>

### CODE REVIEW

File: pkg/restapi/kms/operation/operations.go [hunk 1]
```diff
@@ -304,6 +310,62 @@ func (o *Operation) decryptHandler(rw http.ResponseWriter, req *http.Request) {
...
+
+	provider := prepareKMSProvider(rw, o.provider, keystoreID, request.Passphrase)
+	if provider == nil {
+		return
+	}
+
+	srv := kmsservice.NewService(provider)
+	mac, err := srv.ComputeMAC(keystoreID, keyID, []byte(request.Data))
+	if err != nil {
+		writeErrorResponse(rw, http.StatusInternalServerError, fmt.Sprintf(computeMACFailure, err))
```
Comment (llorllale) [inline pkg/restapi/kms/operation/operations.go:330]: do not leak error details: trustbloc/hub-kms#38

File: pkg/kms/service.go [hunk 2]
```diff
@@ -132,6 +134,26 @@ func (s *service) Decrypt(keystoreID, keyID string, cipher, aad, nonce []byte) (
 	return s.crypto.Decrypt(cipher, aad, nonce, kh)
 }
 
+// ComputeMAC computes message authentication code (MAC) for data.
+func (s *service) ComputeMAC(keystoreID, keyID string, data []byte) ([]byte, error) {
+	kh, err := s.getKeyHandle(keystoreID, keyID)
+	if err != nil {
+		return nil, err
```
Comment (llorllale) [inline pkg/kms/service.go:141]: Please wrap this error

File: pkg/kms/service.go [hunk 3]
```diff
@@ -132,6 +134,26 @@ func (s *service) Decrypt(keystoreID, keyID string, cipher, aad, nonce []byte) (
...
+	}
+
+	return s.crypto.ComputeMAC(data, kh)
+}
+
+// VerifyMAC determines if mac is a correct authentication code (MAC) for data.
+func (s *service) VerifyMAC(keystoreID, keyID string, mac, data []byte) error {
+	kh, err := s.getKeyHandle(keystoreID, keyID)
+	if err != nil {
+		return err
```
Comment (llorllale) [inline pkg/kms/service.go:151]: Please wrap this error

File: pkg/restapi/kms/operation/operations.go (hunk 1 above)
Comment (aholovko) [reply pkg/restapi/kms/operation/operations.go:330]: working on this now, I decided to not put these changes into the current PR given that there is a ticket for refactoring error handling

File: pkg/kms/service.go (hunk 2 above)
Comment (aholovko) [reply pkg/kms/service.go:141]: I am rewriting error handling here (service) and in operations. In this particular case, `getKeyHandle` is a helper function that already wraps an error.

File: pkg/kms/service.go (hunk 3 above)
Comment (aholovko) [reply pkg/kms/service.go:151]: `getKeyHandle` is a helper function that already wraps an error, and also I am working on updating this logic in my next PR

**Pull request merged**